mlfinlab/codependence/correlation.py

```diff
@@ -111,3 +111,51 @@ def distance_correlation(x: np.array, y: np.array) -> float:
     coef = np.sqrt(d_cov_xy) / np.sqrt(np.sqrt(d_cov_xx) * np.sqrt(d_cov_yy))
 
     return coef
+
+
+def kl_dist(corr_A: np.array, corr_B: np.array) -> float:
+    """
+    Returns the Kullback-Leibler distance between two correlation matrices, all elements must be positive.
+
+    Formula used for calculation:
+
+    kl_dist[X, Y] = 0.5 * ( Log( det(Y) / det(X) ) + tr((Y ^ -1).X - n )
+
+    Where n is the dimension space spanned by X.
+
+    Read Don H. Johnson's research paper for more information on Kullback-Leibler distance:
+    `<https://scholarship.rice.edu/bitstream/handle/1911/19969/Joh2001Mar1Symmetrizi.PDF>`_
+
+    :param corr_A: (np.array) Numpy array of the first correlation matrix.
+    :param corr_B: (np.array) Numpy array of the second correlation matrix.
+    :return: (np.float64) the Kullback-Leibler distance between the two matrices.
+    """
+
+    #import scipy.special as sc
+
+    n = corr_A.shape[0]
+    dist = 0.5 * (np.log(np.linalg.det(corr_B) / np.linalg.det(corr_A)) + np.trace(np.linalg.inv(corr_B).dot(corr_A)) - n)
+    return dist
+
+
+def norm_dist(matrix_a: np.array, matrix_b: np.array, r_val=2) -> float:
```

Comment:
Bad name... Please don't use abv

Reply:
Here you are using the strong typing. This is not our convention, despite it being in a few places in the codebase. Did someone ask you add these? (I mean this as a light question)

Reply:
@Jackal08 Codependence was the module with strong typing, as originally developed by @proskurin.

Do you think we should change this?

Reply:
(Change name of the function to `norm_distance`)

Reply:
Hmm I will let you decide. I prefer the other way because it looks better in sphinx but I don't think we have to change it if its a big job.